test/www/jxcore/lib/testUtils.js

```diff
@@ -217,48 +219,72 @@ module.exports.tmpDirectory = tmpDirectory;
  * always resolves with true.
  */
 module.exports.hasRequiredHardware = function () {
-  return new Promise(function (resolve) {
-    if (platform._isRealAndroid) {
-      var checkBleMultipleAdvertisementSupport = function () {
-        Mobile('isBleMultipleAdvertisementSupported').callNative(
-          function (error, result) {
+  function checkNetworks() {
+    if (!ThaliMobile) {
+      ThaliMobile = require('thali/NextGeneration/thaliMobile');
+    }
+    return ThaliMobile.getNetworkStatus()
+      .then(function (networkStatus) {
+        var promises = [];
+        if (networkStatus.wifi === 'off') {
+          promises.push(toggleWifi(true));
```

Comment:
I don't like the idea that a function named `hasRequiredHardware` in fact changes radios state. I would rather see it as 2 seperate functions, for example `hasRequiredHardware` and `enableRequiredHardware`. It requires one extra line in runTests.js, but looks much cleaner